### DISCUSSION TITLE
HasOne/MorphOne loads any fields/relationships when included. Intended or bug?

### DISCUSSION THREAD

**Discussion by marcoraddatz** [laravel/nova-issues#5073] [Q&A] [answered]
- Laravel: 9.40.1
- Nova: 4.19.0

I have a user, who has an address that has logs (user → morphOne address → hasMany logs). Within the user details page, address logs are being show and there's no way to hide them. The same behaviour occurs for fields in general: They cannot be hidden easily (as far as I know). This blows up detail pages heavily.

I don’t know, whether this behaviour is intended or a bug?! If it’s intended, I’d like to suggest a new method to hide a relationship on “include”, such as `hiddenOnRelationships`. If it's a bug, I can create an issue for it.

### User
```
MorphOne::make(__('nova/address.title.singular'), 'address', Address::class),
```

### Address
```
MorphMany::make(__('nova/log.title.plural'), 'logs', Log::class),
```

### Example with relations
![CleanShot 2022-11-16 at 13 39 11@2x](https://user-images.githubusercontent.com/248815/202183894-1f0b6eb3-34c3-4064-a2c7-95df25793789.png)

**Comment by crynobone** [comment, marked as answer]
Yes this is the intended behavior at the moment. However, you can prevent it by using the following:

```php
// before
HasMany::make('Logs'),

// after
$this->mergeWhen(
  $request->isPresentationRequest() && ! $request->viaRelationship(), 
  fn () => [
    HasMany::make('Logs'),
  ]
),
```

**Reply by marcoraddatz** [reply]
Many thanks!
